src/input/tablet.c

```diff
@@ -11,6 +11,29 @@
 #include "input/cursor.h"
 #include "input/tablet.h"
 
+static void
+adjust_for_tablet_area(double tablet_width, double tablet_height,
+		double area_top, double area_right, double area_bottom, double area_left,
+		double *x, double *y)
+{
+	if ((area_top == 0 && area_right == 0 && area_bottom == 0 && area_left == 0)
+			|| tablet_width == 0 || tablet_height == 0) {
+		return;
+	}
+
+	if (area_right > 0 && area_right <= tablet_width) {
+		double area_width = area_right - area_left;
+		double area_width_offset = area_left / tablet_width; // x is in range 0..1
+		*x = (*x - area_width_offset) * tablet_width / area_width;
```

Comment:
I'm a bit confused by the geometry concepts here. Is `x` representing the `1` in `0..1`?
If so, 
1. Is there a better name as `x` makes me think of x0 rather than x1 if you know what I mean.
2. What happens if a user wants to change `left` (in terms of tablet area)?

Reply:
yeah, sorry, that comment is confusing, it was more a note for me that max-x and max-y is always 1. I've converted that into a `const`,  I hope that makes things a bit more readable.

I'm now calculating width/height if it hasn't been specified, with that it also works when just `top` or `left` is specified, thus when only the top of left side should be truncated.